app/app.js

```diff
@@ -597,9 +607,75 @@ app.controller("hemnetCommuterController", ['$scope', '$compile', '$http', '$tim
         }
       }, 1000);
 
+
+      // Check for `active_house_id` in URL query parameters on page load
+      const active_house_id = $location.search().active_house_id;
+
+      if (active_house_id) {
+        // If `active_house_id` exists, simulate a marker click
+        // Assuming we have a method to programmatically trigger the event
+        $scope.simulateClickOnMarker(active_house_id)
+      }
```

Comment:
This sounds promising - but I can't see any code for setting the URL query parameter anywhere?

Reply:
Thank you, Phil, for initiating such a valuable project. I’m glad to contribute some improvements based on my experience using this project while searching for a new apartment.

Regarding the active_house_id, the original scenario was as follows: when a user clicked on a marker, the active_house_id would be updated in the query parameters (line 865). This allowed users to share or copy the URL, enabling others to see the exact house being referenced. However, I eventually removed this feature due to some usability issues. Instead, I decided to add a share button at top of the home information section, alongside the other buttons, but I haven’t had the chance to fully complete it yet.

Reply:
Ok nice, understood! So it's ok to remove these lines of code then I guess?

There is an example on http://angular-ui.github.io/ui-leaflet/examples/ of updating the URL whenever the map moves for sharing, but I never got around to implementing it as I didn't think it was particularly helpful. Agree that a share button would be great though!

Reply:
Note that I pushed quite a bit of new code last night, so make sure you pull in the changes to avoid merge conflicts :)